### ISSUE TITLE
Accessibility issues with input form

### ISSUE THREAD

**Issue by mgifford** [jackocnr/intl-tel-input#884]
We ran into this with a client site, but also in the demo.  This is cleaned up, but:

```
<div class="intl-tel-input allow-dropdown">
   <div class="flag-container">
      <div class="selected-flag" role="combobox" aria-owns="country-listbox" tabindex="0" title="United States: +1">
         <div class="iti-flag us"></div>
         <div class="iti-arrow"></div>
      </div>
      <ul class="country-list hide" id="country-listbox" aria-expanded="false" role="listbox" aria-activedescendant="iti-item-us">
         <li class="country preferred active" id="iti-item-us" role="option" data-dial-code="1" data-country-code="us" aria-selected="true">
            <div class="flag-box">
               <div class="iti-flag us"></div>
            </div>
            <span class="country-name">United States</span><span class="dial-code">+1</span>
         </li>

```

...

```
         <li class="country standard" id="iti-item-ax" role="option" data-dial-code="358" data-country-code="ax">
            <div class="flag-box">
               <div class="iti-flag ax"></div>
            </div>
            <span class="country-name">Åland Islands</span><span class="dial-code">+358</span>
         </li>
      </ul>
   </div>
   <input type="tel" id="phone" class="form-control" autocomplete="off" data-intl-tel-input-id="0" placeholder="(201) 555-0123">
</div>
```

The color list isn't following accessibility best practices.  This should probably be structured as a fieldset.  Certainly the number shouldn't semantically be a div. This is a form after all.

**Comment by jackocnr** [comment]
Hi there, thanks for reaching out. Please could you describe the issues you faced in as much detail as possible. And also give a bit more detail about your proposed changes to make the plugin more accessible. Thanks.

**Comment by mgifford** [comment]
Set the country picker and telephone field in a fieldset, with a legend rather than a label element

The selector code confused IMD tester David: "It doesn't declare its purpose, just the name of whatever country happens to be displayed." It receives keyboard focus before the input field's label ("Telephone"), so declared a country name, cryptically.

More information on fieldsets is here:
https://www.w3.org/TR/WCAG20-TECHS/H65.html

**Comment by mgifford** [comment]
I don't think there is one. It's not a particularly semantic pattern.  Surely though you could just the JS to produce a valid HTML5 form (with the country prefix in a separate input field) but transition that to one input form when it doesn't have focus?

I don't know. I just know that this presently isn't following a good semantic best practice.

**Comment by jackocnr** [comment]
> you could just the JS to produce a valid HTML5 form

Are you saying the current markup produced by the JS is invalid HTML?

> with the country prefix in a separate input field

I think it's a nicer UX for users to be able to type the full number in a single input, country prefix included.

But I also want this plugin to be accessible as possible, so I'm very open to any other suggestions you might have.

**Comment by mgifford** [comment]
Just looked. The div isn't inside the input form, I should have checked. 

But the country list is semantically should be an `<select>` rather than a `<li>`.  

Lists aren't part of forms in any valid HTML.

**Comment by jackocnr** [comment]
It's not a `<select>` because they don't allow you to control the markup/styling of the dropdown, which is a large part of the value of this plugin.

**Comment by mgifford** [comment]
I'm not a css guy, but it seems to be possible https://stackoverflow.com/questions/1895476/how-to-style-a-select-dropdown-with-only-css

**Comment by jackocnr** [comment]
It is possible to style the clickable element on the page, but not the actual dropdown that appears when you click it.

**Comment by mgifford** [comment]
Ok, I haven't quite gotten the image up here, but I'm no css guy:

https://jsfiddle.net/mgifford/xjkcgod8/5/

It looks like it should be possible (if I knew what I was doing).

**Comment by jackocnr** [comment]
Still looks like the standard native dropdown to me on Chrome on MacOS. What browser/platform are you using? Also colours are one thing, but custom markup is another e.g. including an image.

**Comment by mgifford** [comment]
Ok..  I've posted an article here.  We'll see what happens.
https://stackoverflow.com/questions/55383890/how-do-i-add-background-images-the-option-element-of-a-select-input

**Comment by jackocnr** [comment]
Closing due to inactivity.

If you have any other ideas about improving the accessibility, please let me know. Cheers.
